_RFCs/002-node-scaffolding-split.md

```diff
@@ -0,0 +1,60 @@
+- Change Name: node-scaffolding-split
+- Start Date: 2017-11-20
+- RFC PR:
+- Core-Plans Issue:
+
+# Summary
+[summary]: #summary
+
+This RFC proposes splitting the current [scaffolding-node](https://github.com/habitat-sh/core-plans/tree/master/scaffolding-node) into two scaffoldings - one focused on applications that run dynamically and one focused on applications that are compiled and run as static assets.
+
+# Motivation
+[motivation]: #motivation
+
+There are currently two ways of running Node applications. Some applications (especially those created with the [Express](https://expressjs.com/) framework) run the application using Node.js's web server. Others - especially those built with [React](https://reactjs.org/) and [Angular](https://angular.io/) - compile Javascript, etc. as static assets and run them behind a proxy like Nginx. Supporting both of these approaches through one scaffolding would be deeply complex at best and untenable at worst.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This splits the node scaffolding into two different scaffoldings - one for dynamic applications and one for applications that are compiled to static assets.
+
+If you are a node developer creating an Express application (or any other application that uses the Node.js web server) you should use the scaffolding for dynamic applications.
+
+If you are a node developer creating a React or Angular application (or any other application that is compiled into static assets, then run behind a proxy like Nginx) you should use the scaffolding for static applications.
+
+Developers who are using the current node scaffolding need to migrate to either the dynamic scaffolding or the static scaffolding. The dynamic scaffolding will largely be the same as the current node scaffolding, but users will need to change the name of the scaffolding they specify when initializing habitat in their application repos (and in their habitat/plan.sh files).
```

Comment:
This raises the question of whether, since both scaffoldings today are oriented around runtime deps and services, we shouldn't just keep those as they are (so nobody using them would need to change anything), and either make new `-static`/`-build` scaffoldings for the static versions.

Reply:
What do you mean by "both scaffoldings"?

Reply:
Yeah, I had to read it again myself, and I still had to think about it. 😄 I meant the Ruby and Node scaffoldings.

Reply:
Ah!  😄  I'm sticking to just the Node scaffolding in this RFC, but it's good to keep this discussion in mind for the Ruby scaffolding as well.

Reply:
Of course, yeah. I'm only mentioning the Ruby one here in hopes of keeping the behavior of the two as consistent as we can.